client/src/components/CategoriesAccordion.jsx

```diff
@@ -55,6 +69,11 @@ const CategoriesAccordion = ({
                         className="accent-purple"
                         value={subcategory.id}
                         onChange={filterItems}
+                        defaultChecked={
```

Comment:
no need for ternary operator here. 
```suggestion
                        defaultChecked={subcategory.supercategoryId === parseInt(id)}
```
is enough